packages/create-catalyst/src/utils/project-config.ts

```diff
@@ -0,0 +1,33 @@
+import { input } from '@inquirer/prompts';
+import { pathExistsSync } from 'fs-extra/esm';
+import kebabCase from 'lodash.kebabcase';
+import { join } from 'path';
+
+export const projectConfig = async (opts: { projectDir?: string; projectName?: string }) => {
+  let projectName = kebabCase(opts.projectName);
+  let projectDir = opts.projectDir && projectName ? join(opts.projectDir, projectName) : undefined;
+
+  if (projectDir && projectName) {
+    return { projectDir, projectName };
+  }
```

Comment:
🍹 Instead of using `let`'s, we can just do something like this just to clean these files up.
```suggestion
if (opts.projectName && opts.projectDir) {
  return {
    projectDir: join(opts.projectDir, projectName),
    projectName: kebabCase(opts.projectName),
  };
}
```

Reply:
This one is a little tricky, and the more I think about it, I think there's a lot of refactoring that I'm gonna need to do around each option. In this case:
* `projectDir` and `projectName` can be passed as CLI flags
* `projectName` is optional, and if not passed as a flag it would be equal to `undefined` in this function.
* If `projectName` is undefined, we need to **not** return, because we need to prompt the user to enter a project name
* If `projectName` is defined but invalid (such as entering weird characters like `*`), I need to throw an error (just noticed this isn't currently accounted for)
* It's possible for `projectName` to be a space character, in which case I need to throw an error (this isn't accounted for either)
* `projectDir` always defaults to `process.cwd()`, but if the user enters a path that does not exist, I need to throw

I think I can reuse the `validate` method lower down in this file to help with a lot of this. I'll rework it in a separate PR 👍